portal/src/main/webapp/js/src/oncoprint/setup.js

```diff
@@ -997,11 +1024,13 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 		console.log("in setDataType");
 		var def = new $.Deferred();
 		var self = this;
-		QuerySession.getCaseUIDMap().then(function (case_uid_map) {
-		    // TODO: assume multiple studies
-		    var study_id = QuerySession.getCancerStudyIds()[0];
-		    var getUID = function (id) {
-			return case_uid_map[study_id][id];
+		$.when(QuerySession.getCaseUIDMap(), QuerySession.getStudyPatientMap()).then(function(case_uid_map, studyPatientMap) {
+			var getUID = function (studyIdsList,caseId) {
+		    	var tempUIDs = [];
+		    	studyIdsList.forEach(function(studyId){
+		    		tempUIDs.push(case_uid_map[studyId][caseId])
+		    	});
+		    	return tempUIDs;
```

Comment:
this still isn't right - you need to look in the map for a particular study. If two studies share a sample id, this will only return the UID for one of them.